File: src/c/selection_sort.c
#include <stdio.h>

void select_sort(int* p, int size); 
void print_array(int* p, int size); 

int main()
{
	int a[10] = {3, 4, 5, 1, 2, 9, 6, 7, 8, 0}; 

	printf("before sort: "); 
	print_array(a, 10); 

	select_sort(a, 10); 
	printf("after sort : "); 
	print_array(a, 10); 

	return 0; 
}

void select_sort(int* p, int size)
{
	for(int i = 0; i < size - 1; i++)
	{
		int min = i + 1; 
		for(int j = i + 2; j < size; j++)
		{ 
			if(p[j] < p[min])
				min = j; 
		} 
		if(p[min] < p[i])
		{
			int tmp = p[i]; 
			p[i] = p[min]; 
			p[min] = tmp; 
		} 
	}
}

void print_array(int* p, int size)
{
	for(int i = 0; i < size; i++)
		printf("%d ", p[i]); 
	printf("\n"); 
} 

File: src/py/selection_sort.py
def insertion_sort(s): 
	for i in range(len(seq)): 
		min = i
		for j in range(i, len(seq)):
			if seq[j] < seq[min]:
				min = j
		swap(i, min)

def swap(i, j):
	tmp = seq[i]
	seq[i] = seq[j]
	seq[j] = tmp

seq = [7, 10, 5, 3, 8, 4, 2, 9, 6]
insertion_sort(seq)
print seq

